SpatialGDK/Build/Programs/Improbable.Unreal.Scripts/CodeGenerator/Unreal/HeaderGenerator.cs

```diff
@@ -0,0 +1,264 @@
+{name}::Update& Set{Text.SnakeCaseToPascalCase(field.Identifier.Name)}({Types.GetConstAccessorTypeModification(field, bundle, type)});
+"))));
+                }
+
+                if (type.Events.Count > 0)
+                {
+                    builder.AppendLine(Text.Indent(2, string.Join(Environment.NewLine, type.Events.Select(_event => $@"// Event {Text.SnakeCaseToPascalCase(_event.Identifier.Name)} = {_event.EventIndex}
+const {Types.CollectionTypesToQualifiedTypes[Types.Collection.List]}<{Types.GetTypeDisplayName(_event.Type.Type.QualifiedName)}>& Get{Text.SnakeCaseToPascalCase(_event.Identifier.Name)}List() const;
+{Types.CollectionTypesToQualifiedTypes[Types.Collection.List]}<{Types.GetTypeDisplayName(_event.Type.Type.QualifiedName)}>& Get{Text.SnakeCaseToPascalCase(_event.Identifier.Name)}List();
+{name}::Update& Add{Text.SnakeCaseToPascalCase(_event.Identifier.Name)}(const {Types.GetTypeDisplayName(_event.Type.Type.QualifiedName)}&);
+"))));
+                }
+
+                if (type.Fields.Count + type.Events.Count > 0)
+                {
+                    builder.AppendLine(Text.Indent(1, $@"private:"));
+                    if (type.Fields.Count > 0)
+                    {
+                        builder.AppendLine(Text.Indent(2, string.Join(Environment.NewLine, type.Fields.Select(field => $"{Types.CollectionTypesToQualifiedTypes[Types.Collection.Option]}<{Types.GetFieldTypeAsCpp(field, bundle, type)}> _{Text.SnakeCaseToPascalCase(field.Identifier.Name)};"))));
+                    }
+
+                    if (type.Events.Count > 0)
+                    {
+                        builder.AppendLine(Text.Indent(2, string.Join(Environment.NewLine, type.Events.Select(_event => $"{Types.CollectionTypesToQualifiedTypes[Types.Collection.List]}<{Types.GetTypeDisplayName(_event.Type.Type.QualifiedName)}> _{Text.SnakeCaseToPascalCase(_event.Identifier.Name)}List;"))));
+                    }
+                }
+
+                builder.AppendLine(Text.Indent(1, "};"));
+                builder.AppendLine();
+
+                if (bundle.Components[type.QualifiedName].CommandDefinitions.Count > 0)
+                {
+                    builder.AppendLine(Text.Indent(1, $@"class Commands
+{{
+public:
+{Text.Indent(1, string.Join(Environment.NewLine, bundle.Components[type.QualifiedName].CommandDefinitions.Select(command => $@"class {Text.SnakeCaseToPascalCase(command.Identifier.Name)}
+{{
+public:
+{Text.Indent(1, $@"static const Schema_FieldId CommandIndex = {command.CommandIndex};
+struct Request
+{{
+{Text.Indent(1, $@"using Type = {Types.GetTypeDisplayName(command.RequestType.Type.QualifiedName)};
+Request({string.Join($", ", types.Find(t => t.QualifiedName == command.ResponseType.Type.QualifiedName).Fields.Select(f => $"{Types.GetConstAccessorTypeModification(f, bundle, type)} {Text.SnakeCaseToPascalCase(f.Identifier.Name)}"))})
```

Comment:
```suggestion
Request({string.Join($", ", types.Find(t => t.QualifiedName == command.RequestType.Type.QualifiedName).Fields.Select(f => $"{Types.GetConstAccessorTypeModification(f, bundle, type)} {Text.SnakeCaseToPascalCase(f.Identifier.Name)}"))})
```
Sneaky copy pasta error